constraint/pkg/instrumentation/instrumentation.go

```diff
@@ -0,0 +1,24 @@
+// package instrumentation defines primitives to
+// support more observability throughout the contraint framework.
+package instrumentation
+
+// NamedStats covers primitive for many heterogenoues statstics and metrics
+// to be represented to consumers.
+type NamedStats struct {
+	StatsName string
+	Stats     []*Stat
+}
+
+// Stat is a Name, Value, Description tuple that may contain
+// metrics or aggregations of metrics.
+type Stat struct {
+	Name        string
+	Value       string
+	Description string
+}
+
+// StatsEntry comprises of a generalized Key for all associated Stats.
+type StatsEntry struct {
+	Key      interface{}   `json:"key"`
```

Comment:
What is `key` intended for?

Reply:
the `Key` here represents what struct/ object the `NamedStats` list are for.

In our case, we probably will have `template` names but down the road the `key` could be `ConstraintKey` or a `Constraint` name

Reply:
Ah, `key` is probably not descriptive for this use case... something like `kind` or `scope` would be better, though I'm sure there are even more fitting names.

This is in support of some kind of aggregation?

Reply:
Renamed to `Scope`. Per our offline sync up, `Scope` can be the template name or the constraint name (down the road) or some other, more complex key for grouping the stats.